### PR TITLE
added per-model includes support

### PR DESCRIPTION
**Pull request by Xtinct**
Implements #701

### CODE REVIEW

**Comment by ankane** [comment]
Hey @Xtinct, thanks for the PR 👍  As is, this breaks backwards compatibility as `includes` currently accepts a hash for nested associations. I think we'll need to add a new option.

**Comment by Xtinct** [comment]
Hey @ankane, thanks for reviewing. Added option called `includes_per` to handle separate includes.

File: test/test_helper.rb [hunk 1]
```diff
@@ -84,6 +84,14 @@ class Product
     field :alt_description
   end
 
+  class Discount
```
Comment (ankane) [inline test/test_helper.rb]: Can we get away without adding a new model?

File: test/sql_test.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 require_relative "test_helper"
-
```
Comment (ankane) [inline test/sql_test.rb]: Please don't change whitespace.

**Comment by Xtinct** [comment]
Hey @ankane, updated according to your comments.

**Comment by ankane** [comment]
Looks great, thanks @Xtinct 💯  Wasn't super happy with `includes_per_model` (despite suggesting it), so I changed the name to `model_includes`. Will cut a new release with this very soon.

**Pull request merged**